themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js

```diff
@@ -556,10 +556,21 @@ module.controller('RealmLocalizationCtrl', function($scope, Current, $location,
 
     $scope.updateRealmSpecificLocalizationTexts = function() {
         RealmSpecificLocalizationTexts.get({id: realm.realm, locale: $scope.selectedRealmSpecificLocales }, function (updated) {
-            $scope.localizationTexts = updated;
+            $scope.localizationTexts = getSortedArrayByKeyFromObject(updated);
         })
     }
 
+    function getSortedArrayByKeyFromObject(object) {
+        const keys = Object.keys(object).sort();
```

Comment:
Calling `sort()` like this means that the values will be sorted by the UTF-16 code units order value of the string (see [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description)). This means that certain words might not be sorted in what the user would expect, for example if a string contains uppercase letter or special characters (e.g. `réservé` or `RESERVE`).

To mitigate this the String prototype has a [`localeCompare()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) method that can be used to compare strings in a manner that the user would expect them to be ranked in their respective locale.

```suggestion
        const keys = Object.keys(object).sort(function (a, b) {
            return a.localeCompare(b, 'en-US');
        });
```

In this example `en-US` should be replaced with the current locale the user is using in the Admin UI.